src/NativeLog/Record/Record.cpp

```diff
@@ -135,13 +135,13 @@ RecordPtr Record::deserialize(const char * data, size_t size, const SchemaContex
         if (likely(record->codec() == DB::CompressionMethodByte::NONE))
         {
             DB::NativeReader reader(rb, DBMS_TCP_PROTOCOL_VERSION);
-            record->setBlock(reader.read());
+            record->setBlock(reader.read(), /*override_block_info=*/false);
```

Comment:
it seems that deserialized block will lose block info. Probably needs to fix. in future

Reply:
So far we don't carry any information in block.info in a Record when serialization.